core/services/workflows/engine.go

```diff
@@ -863,9 +863,27 @@ func (e *Engine) workerForStepRequest(ctx context.Context, msg stepRequest) {
 		stepStatus = store.StatusCompleted
 	}
 
+	meteringSteps := make([]MeteringReportStep, len(metadata.Metering))
+
+	for idx, detail := range metadata.Metering {
+		unit := MeteringSpendUnit(detail.SpendUnit)
+		value, err := unit.StringToSpendValue(detail.SpendValue)
+		if err != nil {
+			l.Error(fmt.Sprintf("failed to get spend value from %s: %s", detail.SpendValue, err))
+		}
+
+		meteringSteps[idx] = MeteringReportStep{
+			Peer2PeerID: detail.Peer2PeerID,
+			SpendUnit:   unit,
+			SpendValue:  value,
+		}
+	}
+
+	e.meterReport.SetStep(MeteringReportStepRef(stepState.Ref), meteringSteps)
```

Comment:
Hmmmm you don't need to add the execution ID anywhere?

Reply:
I am going to wait for https://github.com/smartcontractkit/chainlink/pull/16923 to be merged before updating this. I think that should satisfy the execution ID requirement, unless you think it should go in the report itself.